src/parameterizations/lateral/MOM_MEKE.F90

```diff
@@ -569,102 +594,116 @@ subroutine step_forward_MEKE(MEKE, h, SN_u, SN_v, visc, dt, G, GV, US, CS, hu, h
       call hchksum(MEKE%MEKE, "MEKE post-update MEKE", G%HI, haloshift=0, scale=US%L_T_to_m_s**2)
     endif
 
-    call cpu_clock_begin(CS%id_clock_pass)
-    call do_group_pass(CS%pass_MEKE, G%Domain)
-    call cpu_clock_end(CS%id_clock_pass)
-
-    ! Calculate diffusivity for main model to use
-    if (CS%MEKE_KhCoeff>0.) then
-      if (.not.CS%MEKE_GEOMETRIC) then
-        if (CS%use_old_lscale) then
-          if (CS%Rd_as_max_scale) then
-            !$OMP parallel do default(shared)
-            do j=js,je ; do i=is,ie
-              MEKE%Kh(i,j) = (CS%MEKE_KhCoeff * &
-                         sqrt(2.*max(0.,barotrFac2(i,j)*MEKE%MEKE(i,j))*G%areaT(i,j)) ) * &
-                         min(MEKE%Rd_dx_h(i,j), 1.0)
-            enddo ; enddo
-          else
-            !$OMP parallel do default(shared)
-            do j=js,je ; do i=is,ie
-              MEKE%Kh(i,j) = CS%MEKE_KhCoeff * &
-                  sqrt(2.*max(0., barotrFac2(i,j)*MEKE%MEKE(i,j))*G%areaT(i,j))
-            enddo ; enddo
-          endif
+  case(EKE_FILE)
+    call time_interp_external(CS%id_eke,Time,data_eke)
+    do j=js,je ; do i=is,ie
+      MEKE%MEKE(i,j) = data_eke(i,j) * G%mask2dT(i,j)
+    enddo; enddo
+    call MEKE_lengthScales(CS, MEKE, G, GV, US, SN_u, SN_v, MEKE%MEKE, depth_tot, bottomFac2, barotrFac2, LmixScale)
+  case(EKE_SMARTREDIS)
+    call pass_vector(u, v, G%Domain)
+    call MEKE_lengthScales(CS, MEKE, G, GV, US, SN_u, SN_v, MEKE%MEKE, depth_tot, bottomFac2, barotrFac2, LmixScale)
+    call infer_meke(G, GV, US, CS%smartredis_meke_CS, Time, MEKE%MEKE, MEKE%Rd_dx_h, u, v, tv, h, dt)
+  case default
+    call MOM_error(FATAL,"Invalid method specified for calculating EKE")
+  end select
+
+  call cpu_clock_begin(CS%id_clock_pass)
+  call do_group_pass(CS%pass_MEKE, G%Domain)
+  call cpu_clock_end(CS%id_clock_pass)
+
+  ! Calculate diffusivity for main model to use
+  if (CS%MEKE_KhCoeff>0.) then
+    if (.not.CS%MEKE_GEOMETRIC) then
+      if (CS%use_old_lscale) then
+        if (CS%Rd_as_max_scale) then
+          !$OMP parallel do default(shared)
+          do j=js,je ; do i=is,ie
+            MEKE%Kh(i,j) = (CS%MEKE_KhCoeff * &
+                       sqrt(2.*max(0.,barotrFac2(i,j)*MEKE%MEKE(i,j))*G%areaT(i,j)) ) * &
+                       min(MEKE%Rd_dx_h(i,j), 1.0)
+          enddo ; enddo
         else
           !$OMP parallel do default(shared)
           do j=js,je ; do i=is,ie
             MEKE%Kh(i,j) = CS%MEKE_KhCoeff * &
-                sqrt(2.*max(0., barotrFac2(i,j)*MEKE%MEKE(i,j))) * LmixScale(i,j)
+                sqrt(2.*max(0., barotrFac2(i,j)*MEKE%MEKE(i,j))*G%areaT(i,j))
```

Comment:
I'm not sure if GH is reporting this right. Not sure why adding SmartRedis involves replacing LmixScale with sqrt(areaT)?

Reply:
Indeed, it looks like the whitespce changes are confusing GH here.